src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -67,6 +85,76 @@ public FilterPanel(UI ui, IModel model, PanelControl parentPanelControl, int pan
 
         filterTextField.setKeywords(all);
     };
+    
+    private Node createPanelMenuBar() {
+        menuBarNameArea = createNameArea();
+        HBox menuBarCloseArea = createCloseButton();
+        
+        panelMenuBar = new HBox();
+        panelMenuBar.setSpacing(2);
+        panelMenuBar.setMinWidth(PANEL_WIDTH);
+        panelMenuBar.setMaxWidth(PANEL_WIDTH);
+        panelMenuBar.getChildren().addAll(menuBarNameArea, menuBarCloseArea);
+        panelMenuBar.setPadding(new Insets(0, 0, 8, 0));
+        return panelMenuBar;
+    }
+    
+    private HBox createNameArea() {
+        HBox nameArea = new HBox();
+        
+        nameText = new Text(panelName);
+        nameText.setId(model.getDefaultRepo() + "_col" + panelIndex + "_nameText");
+        
+        nameBox = new HBox();
+        nameBox.getChildren().add(nameText);
+        nameBox.setMinWidth(330);
+        nameBox.setMaxWidth(330);
+        nameBox.setAlignment(Pos.CENTER_LEFT);
+        nameBox.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent mouseEvent) {
+                if (mouseEvent.getButton().equals(MouseButton.PRIMARY)) {
+                    if (mouseEvent.getClickCount() == 2) {
+                        mouseEvent.consume();
```

Comment:
This can be replaced with a lambda.

Also some nesting can be removed from the conditional:

```java
boolean isDoubleClick = ... && ...;
if (isDoubleClick) {
    ...
}
```

In future we can factor this out further maybe, since double clicks are fairly commonplace.